src/main/scala/org/broadinstitute/dsde/agora/server/ServerInitializer.scala

```
@@ -30,42 +30,62 @@ class ServerInitializer extends LazyLogging {
 
-  private def startWebServiceActors() = {
+  private def startWebService() = {
     implicit val bindTimeout: Timeout = 120.seconds
-    val service = actorSystem.actorOf(ApiServiceActor.props(permsDataSource, healthMonitor), "agora-actor")
-    Await.result(IO(Http) ? Http.Bind(service, interface = AgoraConfig.webserviceInterface, port = AgoraConfig.port), bindTimeout.duration) match {
-      case CommandFailed(b: Http.Bind) =>
-        logger.error(s"Unable to bind to port ${AgoraConfig.port} on interface ${AgoraConfig.webserviceInterface}")
-        actorSystem.terminate()
-        stopAndExit()
-      case _ => logger.info("Actor system started.")
+
+    val apiService = new ApiService(permsDataSource, healthMonitor)
+
+    Http().bindAndHandle(apiService.route, "0.0.0.0", 8000)
+      .recover {
+        case t: Throwable =>
+          logger.error(s"Unable to bind to port ${AgoraConfig.port} on interface ${AgoraConfig.webserviceInterface}")
+          actorSystem.terminate()
+          stopAndExit()
+          throw t
+      }
+  }
+
+  /**
+    * Firecloud system maintains its set of admins as a google group.
+    * If such a group is specified in config, poll it at regular intervals
+    * to synchronize the admins defined in our users table.
+    */
+  private def startAdminGroupPoller() = {
+    AgoraConfig.adminGoogleGroup match {
+      case Some(group) =>
+        val adminGroupPoller = actorSystem.actorOf(AdminSweeper.props(AdminSweeper.adminsGoogleGroupPoller, permsDataSource))
+        adminGroupPollerSchedule = actorSystem.scheduler.schedule(5 seconds, AgoraConfig.adminSweepInterval minutes, adminGroupPoller, Sweep)
+      case None =>
     }
   }
 
-  private def stopWebServiceActors() {
-    IO(Http) ! Http.CloseAll
+  private def stopAdminGroupPoller() = {
+    Try(adminGroupPollerSchedule.cancel()) recover {
+      case e: NullPointerException => // Nothing to do; no scheduler was created at the first place
+      case t: Throwable => throw t
```

since this is only used during shutdown, I say write a log message and swallow the error (don't rethrow), avoiding an ugly stack trace. Not a big deal; not a blocker for this PR.